connectors/sources/jira.py

```diff
@@ -0,0 +1,496 @@
+            )
+        else:
+            auth = self.configuration["username"], self.configuration["password"]
+
+        basic_auth = aiohttp.BasicAuth(
+            login=auth[0],
+            password=auth[1],
+        )
+        request_headers = {
+            "accept": "application/json",
+            "content-type": "application/json",
+        }
+        timeout = aiohttp.ClientTimeout(total=None)  # pyright: ignore
+        self.session = aiohttp.ClientSession(
+            auth=basic_auth,
+            headers=request_headers,
+            timeout=timeout,
+            raise_for_status=True,
+        )
+
+    async def close(self):
+        """Closes unclosed client session"""
+        self._sleeps.cancel()
+        if self.session is None:
+            return
+        await self.session.close()
+        self.session = None
+
+    async def _api_call(self, url_name, **url_kwargs):
+        """Make a GET call for Atlassian API using the passed url_name with retry for the failed API calls.
+
+        Args:
+            url_name (str): URL Name to identify the API endpoint to hit
+            url_kwargs (dict): Url kwargs to format the query.
+
+        Raises:
+            exception: An instance of an exception class.
+
+        Yields:
+            response: Return api response.
+        """
+        retry = 0
+        url = parse.urljoin(self.host_url, URLS[url_name].format(**url_kwargs))
+        while True:
+            try:
+                async with self.session.get(  # pyright: ignore
+                    url=url,
+                    ssl=self.ssl_ctx,
+                ) as response:
+                    yield response
+                    break
+            except Exception as exception:
+                if isinstance(
+                    exception,
+                    ServerDisconnectedError,
+                ):
+                    await self.close()
+                    self._generate_session()
+                retry += 1
+                if retry > self.retry_count:
+                    raise exception
+                logger.warning(
+                    f"Retry count: {retry} out of {self.retry_count}. Exception: {exception}"
+                )
+                await self._sleeps.sleep(RETRY_INTERVAL**retry)
+
+    async def _paginated_api_call(self, url_name):
+        """Make a paginated API call for Jira objects using the passed url_name with retry for the failed API calls.
+
+        Args:
+            url_name (str): URL Name to identify the API endpoint to hit
+
+        Yields:
+            response: Return api response.
+        """
+        start_at = 0
+
+        should_paginate = True
+        while should_paginate:
+            async for response in self._api_call(
+                url_name=url_name, start_at=start_at, max_results=FETCH_SIZE
+            ):
+                response_json = await response.json()
+                total = response_json.get("total")
```

Comment:
if `total` does not exists you get `None` and the next test will fail. 

if you always get `total` and it's an int, change this to:

```
total = response_json["total"]
```

if not, make that code more robust:

```
total = int(response_json.get("total", 0))
```